ogc: fix wrong values reported on GameCube controller triggers

mardy opened the pull request:
These two axes values are unsigned, and all 8 bits can be used; therefore, in order to reach SHRT_MAX we only need to shift them by 7 bits, not 8. This avoids having the value wrap down to zero midway.

WinterMute commented:
I honestly think we should just consolidate these files rather than repeating the code like this.

mardy commented:
I will try to consolidate the two files for SDL2 (there will be quite a few `ifdef`s...), then we can see if it's worth to do the same for SDL1.2.

But in any case, this is the code we have right now, and it's wrong :-)

mardy committed:
ogc: fix wrong values reported on GameCube controller triggers

These two axes values are unsigned, and all 8 bits can be used;
therefore, in order to reach SHR_MAX we only need to shift them by 7
bits, not 8. This avoids having the value wrap down to zero midway.
